app/models/project_step.rb

```diff
@@ -42,6 +45,9 @@ class NoChildrenAllowedError < StandardError; end
   SUPER_EARLY_PERIOD = 7.0 # days
   SUPER_LATE_PERIOD = 30.0 # days
 
+  belongs_to :schedule_ancestor,   class_name: 'ProjectStep'
+  has_many   :schedule_decendants, class_name: 'ProjectStep', foreign_key: :schedule_ancestor_id
```

Comment:
I'd prefer we not do tabular style like this. Then if we have to change lines later the changes might cause changes in other unrelated lines just to maintain the tabular style.